ecr-login/api/client.go

```diff
@@ -98,6 +99,22 @@ func (self *defaultClient) GetCredentials(registry, image string) (*Auth, error)
 	return nil, fmt.Errorf("No AuthorizationToken found for %s", registry)
 }
 
+func (self *defaultClient) ListCredentials() ([]*Auth, error) {
+	log.Debug("Listing Credentials from cache")
+
+	auths := []*Auth{}
+	for _, authEntry := range self.credentialCache.List() {
```

Comment:
I *think* the best thing to do here is to just return everything then.  The Docker CLI will take care of requesting credentials by calling get at that point, which will handle returning the appropriate credentials.

If you want to scope it down further, you could look at by account.  But that will require making a call to [GetCallerIdentity](http://docs.aws.amazon.com/STS/latest/APIReference/API_GetCallerIdentity.html) that you might not have permission to make, at which point you'll probably want to fall-back to returning everything anyway.